src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -109,20 +119,18 @@ public bool SnippetShortcutExists_NonBlocking(string shortcut)
         public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
             => false;
 
-        private void PopulateSnippetCaches()
+        private async Task PopulateSnippetCachesAsync()
```

Comment:
What would this look like if we renamed this to QueuePopulateSnippetCaches, it _didn't_ return Task, and then also did the async listener stuff inside of this? Right now every one of the callers has do the same BeginAsyncOperation/CompletesAsyncOperation/Forget dance, and seems we can put that into one single place.

Reply:
I thought about that, my only concern was that it would then not include the bits switching to the main thread and retrieving the services during initialization.  So then we could have the object be constructed, but the async operation thinks nothing is in progress (if for example we paused on switching to the main thread on line 68)